src/main/scala/com/comcast/xfinity/sirius/uberstore/segmented/SegmentedCompactor.scala

```diff
@@ -15,15 +15,28 @@
  */
 package com.comcast.xfinity.sirius.uberstore.segmented
 
+import com.comcast.xfinity.sirius.api.SiriusConfiguration
+import com.comcast.xfinity.sirius.api.impl.{Delete, OrderedEvent}
+
 import scalax.file.Path
 import java.io.File
 import annotation.tailrec
 
 object SegmentedCompactor {
-
   val COMPACTING_SUFFIX = ".compacting"
   val TEMP_SUFFIX = ".temp"
 
+  def apply(siriusConfig: SiriusConfiguration) : SegmentedCompactor = {
+
+    val maxDeleteAgeHours = siriusConfig.getProp(SiriusConfiguration.COMPACTION_MAX_DELETE_AGE_HOURS, Integer.MAX_VALUE) // a really long time by default
+    val maxDeleteAgeMillis = 60L * 60 * 1000 * maxDeleteAgeHours
+
+    new SegmentedCompactor(maxDeleteAgeMillis)
+  }
+}
+
+class SegmentedCompactor(maxDeleteAgeMillis: Long) {
```

Comment:
This can probably made at least package private. Outside of tests, the apply() method should be used to instantiate this class.